### PR TITLE
Cli tool anonymizer

### PR DESCRIPTION
**Pull request by VukW**
sub-PR for https://github.com/mlcommons/GaNDLF/pull/818 : adding `gandlf anonymizer` subcommand

- [x] Code is here, both new way (via `gandlf` cli tool) and old way (via `gandlf_*` entrypoint) are supported
- [x] Tests are here
- [x] Docs are fixed
- [x] Code style is blacked

### CODE REVIEW

File: GANDLF/entrypoints/anonymizer.py [hunk 1]
```diff
@@ -1,12 +1,83 @@
...
+def _anonymize_images(
+    input_dir: str, output_file: str, config_path: Optional[str], modality: str
+):
+    input_dir = os.path.normpath(input_dir)
+    output_file = os.path.normpath(output_file)
+    # TODO: raise an error if config pass provided but not exist (user made a typo?)
+    if config_path and os.path.isfile(config_path):
+        config = yaml.safe_load(open(config_path, "r"))
+    else:
+        config = None
```
Comment (sarthakpati) [inline GANDLF/entrypoints/anonymizer.py]: This is just to improve code coverage

```suggestion
    config = None
    if config_path and os.path.isfile(config_path):
        config = yaml.safe_load(open(config_path, "r"))

    assert config is not None, "A valid config file should be passed"
```

**Review by sarthakpati — CHANGES_REQUESTED** [review]
Minor comment but looks good otherwise!

File: GANDLF/entrypoints/anonymizer.py (hunk 1 above)
Comment (VukW) [reply GANDLF/entrypoints/anonymizer.py]: Are you sure we want to require config file from users?
Actually, according to the [anonymizer code](https://github.com/mlcommons/GaNDLF/blob/master/GANDLF/anonymize/__init__.py#L21) , it's ok if config is empty / not passed at all

Comment (sarthakpati) [reply GANDLF/entrypoints/anonymizer.py]: Got it, thanks for the reminder. In that case, I think the `assert` portion of my suggestion can be removed. That should be the same as before, right? 😄

Comment (VukW) [reply GANDLF/entrypoints/anonymizer.py]: Right! Committed changes:)

**Review by sarthakpati — APPROVED** [review]

**Review by sarthakpati — APPROVED** [review]

**Pull request closed without merging**